helix-core/src/lib.rs

```diff
@@ -68,6 +68,7 @@ pub fn find_root(root: Option<&str>, root_markers: &[String]) -> std::path::Path
             .any(|marker| ancestor.join(marker).exists())
         {
             top_marker = Some(ancestor);
+            break;
```

Comment:
Now that we're always searching eagerly in this loop, we can get rid of the `top_marker` variable and return immediately if we see a marker or a `.git`:

```rs
for ancestor in root.ancestors() {
    if root_markers.iter().any(|marker| marker.ancestor.join(marker).exists) || ancestor.join(".git").is_dir() {
        return ancestor.to_path_buf();
    }
}
```

returning just `current_dir` at the bottom of the function